File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter
def solution(participant, completion):
    answer = ''
    
    dicP = Counter(participant)
    dicC = Counter(completion)
    
    for p in dicP.keys():
        if p not in dicC or dicP[p] != dicC[p]:
            return p
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        HashMap<String, Integer> map = new HashMap<>();
        
        for (String parti : participant) {
            map.put(parti, map.getOrDefault(parti, 0) + 1);
        }
        for (String compl : completion) {
            map.put(compl, map.get(compl) - 1);
        }
        
        for (String parti : participant) {
            if (map.get(parti) >= 1) {
                return parti;
            }
        }
        
        return answer;
    }
}